src/glados/schema_tags_generator.py

```diff
@@ -173,16 +173,22 @@ def get_schema_obj_for_compound(chembl_id, request):
     if item_name is not None:
         metadata_obj['name'] = item['pref_name']
 
+    iupac_name = ""
     alternate_names_set = set()
     raw_synonyms = item['molecule_synonyms']
     if raw_synonyms is not None:
         for raw_syn in raw_synonyms:
             alternate_names_set.add(raw_syn['molecule_synonym'])
+            if raw_syn['syn_type'] == 'SYSTEMATIC':
+                iupac_name = raw_syn['molecule_synonym']
```

Comment:
if the IUPAC name is found there is no need to continue iterating put a break to escape from the for. However, worst case scenario there should be less than 1000 synonyms, so it is not crucial.

Reply:
But the for is used for filling the alternate_names_set, I hijacked it for that conditional. In theory there should be only one alternate name of the type "systematic"